src/ql/com/topology.cc

```diff
@@ -685,42 +689,73 @@ utils::UInt Topology::get_distance(Qubit source, Qubit target) const {
         }
         return d;
     }
-
     return distance[source][target];
 }
 
 /**
  * Returns the distance between the given two qubits in terms of cores.
  */
 utils::UInt Topology::get_core_distance(Qubit source, Qubit target) const {
-    if (get_core_index(source) == get_core_index(target)) return 0;
+    if (source == target ||
```

Comment:
If source == target, then logically get_core_index(source) == get_core_index(target), therefore isn't that condition useless? I think previous version was OK no?

Reply:
Well, my point here was that it saves you from calling a couple of functions. But, on the other hand, it is true that it adds an extra check, and it is an extra check that is usually not recommended, like the typical check for self assingment in an assignment operator (we shouldn't check for the unprobable case).